pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
```

Comment:
It would be better to extract this to its own method, and only mutate the field for the duration of the validate call, not permanently.  It would also be better to use a valid value for image, rather than relying on the empty whitespace being treated for the field.

I.e.:

```
func setEmptyImageChangeReferences(template, triggers) sets.String {
  // ... mutate each container with a trigger, return the names of those containers
  ... 
    containers[i].Image = "unset"
  ...
}
func resetEmptyImageChangeReferences(template, empty) {
  for i := range template.Containers {
    if empty.Has(template.Containers[i].Name) {
      template.Containers[i].Image = ""
    }
  }
}

func validateDeploymentConfigPodSpecTemplate(...) {
  emptyImages := setEmptyImageChangeReferences(template, triggers)
  errs := validatePodSpecTemplate(template)
  resetEmptyImageChangeReferences(template, emptyImages)
  return errs
}
```

We should not let our mutation escape the validation method.